### PR TITLE
include rocsparse library when compiling with Hypre compiled with hip

### PR DESCRIPTION
**Pull request by etpalmer63**
## Summary
Includes the rocsparse library for HIP enabled builds with HYPRE 

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
We might want to move this into `Make.hypre` just like what we do to CUDA.

**Comment by etpalmer63** [comment]
:+1:

File: Tools/CMake/AMReXParallelBackends.cmake [hunk 1]
```diff
@@ -272,7 +272,7 @@ if (AMReX_HIP)
    target_compile_options(amrex PUBLIC $<$<COMPILE_LANGUAGE:CXX>:-m64>)
 
    # ROCm 4.5: use unsafe floating point atomics, otherwise atomicAdd is much slower
-   # 
+   #
```
Comment (ax3l) [inline Tools/CMake/AMReXParallelBackends.cmake:275]: See above: `rocrand` et al. or `roctx` as examples.
Guard with `if (AMReX_HYPRE)`.

rocSPARSE refs.:
- https://github.com/ROCmSoftwarePlatform/rocSPARSE
- https://rocsparse.readthedocs.io

**Comment by ax3l** [comment]
General question: why is that needed? Does Hypre expose rocSPARSE in public headers or forgets to add transitive link targets?

**Comment by etpalmer63** [comment]
The need for this may have been an error on my part. I need to do some more testing to be sure though. -- Thanks for your patience :pray:

**Comment by etpalmer63** [comment]
I got AMReX with Hypre and MPI to run on crusher without any of these changes, so its clear they are not needed. Closing this PR.

**Pull request closed without merging**